### PR TITLE
Webpage update

### PR DESCRIPTION
**Pull request by justin-richling**
This implements the basic upgraded website layout after suggestions in Discussions #114. A sample can be viewed [here](https://project.cgd.ucar.edu/projects/ADF/web_update/f.c6_3_41.FWscHIST.ne30_L58_exner.001_vs_f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001_1980_1989/index.html)

### Major feature upgrades:

This layout will provide easier user interface and experience such that all plots appear to be on the same page so that the user doesn't have to navigate away to cycle through different variables/plot types/seasons. 

This PR will also add the AMWG comparison tables for model vs model runs. 

The NH and SH Polar plots have been added as well.

The Lat/Lon plots have been reordered such that the two case plots are on one row and the difference plot is below similar to the layout of the Polar plots. This will save space on the page and ensure the plots won't get cut off the browser.

Under each plot type, the available variables have been broken up into categories based off the CAM7 [spreadsheet](https://docs.google.com/spreadsheets/d/17IxS8Du8wkLWefzELex3e3RCr3GrpSr6o5D3GZm8xto/edit?usp=sharing).

Case names in the header have been changed from ```Case 1``` and ```Case 2``` to ```Test Case``` and ```Control Case```.

Table navigation has been added to the main navigation bar at top of page when under the Table plot type category.

If there are many variable categories, the current plots will scroll down the page when the user scrolls.

Clicking on any plot image will bring it up in a new browser tab for zoomable option.

### Files changed:

- ```lib/adf_diag.py```
- ```lib/plotting_functions.py```
- ```lib/website_templates/adf_diag.css```
- ```lib/website_templates/template.html```
- ```lib/website_templates/template_index.html```
- ```lib/website_templates/template_mean_diag.html```
- ```lib/website_templates/template_mean_diag_LatLon.html```
- ```lib/website_templates/template_mean_diag_Zonal.html```
- ```lib/website_templates/template_mean_table.html```
- ```scripts/analysis/amwg_table.py```
- ```scripts/plotting/polar_map.py```
- ```scripts/plotting/zonal_mean.py```

### Files added:

- ```lib/website_templates/template_mean_diag_NHPolar.html```
- ```lib/website_templates/template_mean_diag_SHPolar.html```

### Notes:

The layout is still open to change, but this should be a good start to have user test out.

The Lat/Lon plots seem to be a little small, this is something we can work on after the fact.

AMWG Tables are still on a separate page until we move forward with being able to pull data frame data between scripts.

### CODE REVIEW

**Comment by nusbaume** [comment]
@justin-richling Would you be able to bring in the vector plots as well to this new website layout?  If that is going to be a heavy lift and/or you want some help just let me know.  Also, if you want to test the vector plots then just add `TAUX` and `TAUY` to your variable list and the ADF should generate vector plots for those quantities.  Thanks!

**Comment by justin-richling** [comment]
@nusbaume Vector plots have been successfully integrated, example [here](https://project.cgd.ucar.edu/projects/ADF/web_update/vector_test/f.e21.FHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf_1980_1989_vs_f.e21.FWscHIST.ne30_L48_BL10_cam6_3_019_plus_CESM2.2.001_zm2.hf_1980_1989/html_img/plot_page_Surface_Wind_Stress_SON_LatLon_Vector.html). I also fixed a quick bug that was keeping the plot image html files from being populated (probably got changed with all the newest PR's) but it's all good now and should be ready for a final look over and merge (fingers crossed!)

File: scripts/analysis/amwg_table.py [hunk 1]
```diff
@@ -295,17 +302,39 @@ def _spatial_average(indata):
 
 #####
 
-def _write_html(f, out):
+def _write_html(f, out,case_name,case_name_list):
+    if case_name == case_name_list[0]:
+        case = "Test"
+    else:
+        case = "Control"
```
Comment (nusbaume) [inline scripts/analysis/amwg_table.py:309]: In multi-case ADF runs the last entry in the case names list will be the baseline, and everything else will be a test case. I am not sure how you want to handle it at the moment (given that this function doesn't appear ready for multi-case usage yet?), but I would recommend setting the "Control" case to only be when `case_name = case_name_list[-1]`.

File: scripts/analysis/amwg_table.py [hunk 2]
```diff
@@ -295,17 +302,39 @@ def _spatial_average(indata):
...
     import pandas as pd
     df = pd.read_csv(f)
     html = df.to_html(index=False, border=1, justify='center', float_format='{:,.3g}'.format)  # should return string
-    preamble = f"""<html><head></head><body><h1>{f.stem}<h1>"""
+    #preamble = f"""<html><head></head><body><h1>{f.stem}<h1>"""
```
Comment (nusbaume) [inline scripts/analysis/amwg_table.py:313]: Remove commented-out code?

File: scripts/analysis/amwg_table.py [hunk 3]
```diff
@@ -295,17 +302,39 @@ def _spatial_average(indata):
...
+          <ul class="dropdown">
+            <li><a href="https://www.cesm.ucar.edu">CESM</a></li>
+            <li><a href="https://www.cesm.ucar.edu/working_groups/Atmosphere/?ref=nav">AMWG</a></li>
+            <li><a href="https://www.cgd.ucar.edu/amp/">AMP</a></li>
+          </ul>
+        </li>
+        <li><a href="https://github.com/NCAR/ADF">About</a></li>
+        <li><a href="https://github.com/NCAR/ADF/discussions">Contact</a></li>
+      </ul>
+    </nav><h1>AMP Diagnostics</h1><h2>{case} Case: {f.stem}<h2>"""
```
Comment (nusbaume) [inline scripts/analysis/amwg_table.py:330]: I might change this header to be something like `Model Diagnostics`, `CAM Diagnostics`, or `ADF Diagnostics`.

File: scripts/analysis/amwg_table.py [hunk 4]
```diff
@@ -326,19 +355,33 @@ def _df_comp_table(write_html,output_location,case_names):
...
+          <ul class="dropdown">
+            <li><a href="https://www.cesm.ucar.edu">CESM</a></li>
+            <li><a href="https://www.cesm.ucar.edu/working_groups/Atmosphere/?ref=nav">AMWG</a></li>
+            <li><a href="https://www.cgd.ucar.edu/amp/">AMP</a></li>
+          </ul>
+        </li>
+        <li><a href="https://github.com/NCAR/ADF">About</a></li>
+        <li><a href="https://github.com/NCAR/ADF/discussions">Contact</a></li>
+      </ul>
+    </nav><h1>AMP Diagnostics</h1><h2>AMWG Case Comparison</h2><h2>Test Case: {case_names[0]}<br/>Control Case: {case_names[-1]}</h2>"""
```
Comment (nusbaume) [inline scripts/analysis/amwg_table.py:383]: I might change this header to be something like `Model Diagnostics`, `CAM Diagnostics`, or `ADF Diagnostics`.

File: scripts/plotting/polar_map.py [hunk 5]
```diff
@@ -241,8 +241,8 @@ def polar_map(adfobj):
                             shfig = make_polar_plot(mseasons[s], oseasons[s], dseasons[s], hemisphere="SH", **vres)
 
                             # Assign titles to each figure:
-                            nhfig.suptitle(f"{s} - {case_name} and {data_name}")
-                            shfig.suptitle(f"{s} - {case_name} and {data_name}")
+                            #nhfig.suptitle(f"{s} - {case_name} and {data_name}")
+                            #shfig.suptitle(f"{s} - {case_name} and {data_name}")
```
Comment (nusbaume) [inline scripts/plotting/polar_map.py:245]: Remove commented-out code?

File: scripts/plotting/polar_map.py [hunk 6]
```diff
@@ -396,8 +396,8 @@ def make_polar_plot(d1:xr.DataArray, d2:xr.DataArray, difference:Optional[xr.Dat
     lons, lats = np.meshgrid(lon_cyclic, d1.lat)
 
     fig = plt.figure(figsize=(10,10))
-    gs = gridspec.GridSpec(2, 4)
-    gs.update(wspace=0.9)
+    gs = gridspec.GridSpec(2, 4, wspace=0.9)
+    #gs.update(wspace=0.9)
```
Comment (nusbaume) [inline scripts/plotting/polar_map.py:400]: Remove commented-out code?

File: scripts/plotting/polar_map.py [hunk 7]
```diff
@@ -459,7 +459,7 @@ def make_polar_plot(d1:xr.DataArray, d2:xr.DataArray, difference:Optional[xr.Dat
                     borderpad=0,
                     )
     fig.colorbar(img3, cax=cb_diff_ax)
-    fig.suptitle("SEASON IN TITLE")
+    #fig.suptitle("SEASON IN TITLE")
```
Comment (nusbaume) [inline scripts/plotting/polar_map.py:462]: Remove commented-out code?

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good!  However, I do have some change requests (mostly related to code-cleanup).

**Comment by justin-richling** [comment]
@nusbaume I think I've gone through all your comments and have updated the branch. I just did a quick test with no issues so when you get a chance to it test on your end, let me know if anything else needs to be addressed.

Thanks again,

**Review by nusbaume — APPROVED** [review]

**Pull request merged**